local.go

```diff
@@ -54,59 +54,100 @@ func Local(f Function, initX []float64, settings *Settings, method Method) (*Res
 	}
 
 	startTime := time.Now()
-
 	funcs, funcInfo := getFunctionInfo(f)
+	status := NotTerminated
+
+	if funcInfo.IsStatuser {
+		_, err := funcs.status.Status()
+		if err != nil {
+			// Function status is bad, panic.
+			// TODO: Document this panic in the function comment.
+			panic(fmt.Sprintf("bad initial function status: %v", err))
```

Comment:
Honestly though, I'm okay with not having this call at all. Right now the flow is simple: 1) Call the function to generate the initial location if necessary. 2) Check the status before each step of the minimization, returning the Status and error. Status is checked before the first step of the minimization, so all this does is catch an error if the initial condition is not specified. If I were a function writer, I would have the returned function value be NaN or Inf if the function was actually in error, both of which will create an error if no initial condition is specified. If the initial condition is specified, we already check Status before the first step and do the reasonable thing.

If you do want this check here, it should behave the same as all the other checks (i.e. go to Cleanup).